packages/cloud_firestore/cloud_firestore/test/query_test.dart

```diff
@@ -309,6 +336,20 @@ void main() {
         expect(q.parameters['endBefore'], isNull);
         expect(q.parameters['endAt'], equals(['456']));
       });
+
+      test('startAfterDocument() encode DocumentReference to Platform class',
+          () async {
+        DocumentSnapshot doc =
+            await firestore.collection('/movies').doc('exists').get();
+        Query q = query!.orderBy('reference').startAfterDocument(doc);
+        expect(
+          q.parameters['startAfter'].first,
+          allOf([
+            isA<MethodChannelDocumentReference>(),
+            isA<DocumentReferencePlatform>(),
+          ]),
+        );
+      });
```

Comment:
Could you move this test to the integration tests: `packages/cloud_firestore/cloud_firestore/example/integration_test/query_e2e.dart`? So you don't need to mock the channel and we can check it works on a real device.